ehr/reconcile/planner.go

```diff
@@ -0,0 +1,87 @@
+package reconcile
+
+import (
+	"context"
+
+	duration "github.com/xhit/go-str2duration/v2"
+
+	"github.com/tidepool-org/platform/clinics"
+	"github.com/tidepool-org/platform/ehr/sync"
+	"github.com/tidepool-org/platform/log"
+	"github.com/tidepool-org/platform/task"
+)
+
+type Planner struct {
+	clinicsClient clinics.Client
+	logger        log.Logger
+}
+
+func NewPlanner(clinicsClient clinics.Client, logger log.Logger) *Planner {
+	return &Planner{
+		clinicsClient: clinicsClient,
+		logger:        logger,
+	}
+}
+
+func (p *Planner) GetReconciliationPlan(ctx context.Context, syncTasks map[string]task.Task) (*ReconciliationPlan, error) {
+	toCreate := make([]task.TaskCreate, 0)
+	toDelete := make([]task.Task, 0)
+	toUpdate := make(map[string]*task.TaskUpdate)
+
+	// Get the list of all EHR enabled clinics
+	clinicsList, err := p.clinicsClient.ListEHREnabledClinics(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// At the end of the loop syncTasks will contain only the tasks that need to be deleted,
+	// and toCreate will contain tasks for new clinics that need to be synced.
+	for _, clinic := range clinicsList {
+		clinicId := *clinic.Id
+		settings, err := p.clinicsClient.GetEHRSettings(ctx, clinicId)
+		if err != nil {
+			return nil, err
```

Comment:
The `clinicsClient` appears to be generated code that doesn't return platform errors. Therefore you might consider if these should use `errors.Wrap`. (I won't mention this any more for this PR, but there are other places I haven't pointed out.)

Reply:
Can you please clarify what are the benefits of using the platform `errors.Wrap`, because I'm not familiar with those.

Reply:
I've been using the standard library most recently. This may be incorrect, but it's not obvious or documented why the platform errors package should be used instead, hence my question.

Reply:
Hmm, what do folks think about me writing a "Guide to Writing Consistent Code in the Platform Repo"? 😉 

Yes, there is a reason to use the Platform `errors` package and, more explicitly, the `errors.Wrap` functionality. The `errors` package adds a lot more context to the error that chains correctly with multiple wraps. Now, while the **current** Golang `errors` package does *some* of that, the rest of Platform isn't built to handle that in any way.

So, in short, if you want errors to propagate correctly and do the "right thing" in the rest of Platform, then use the Platform `errors` package and use `errors.Wrap` when you want to add further context or explanation to the error.

I'm fine if we want to intentionally revisit this and make appropriate changes, but IMO, we should stick with how it is currently done so it works and is written as expected.

My 2 cents...

Reply:
Thanks for the explanation - I'll make the change, because those errors might eventually bubble up to the task error handler.